virtual-display-driver-cli/src/main.rs

```diff
@@ -0,0 +1,343 @@
+
+        /// Refresh rate for the new mode. Pass multiple times to support
+        /// multiple refresh rates in this mode.
+        #[clap(short, long, default_value = "60")]
+        refresh_rate: Vec<driver_ipc::RefreshRate>,
+    },
+    /// Remove a resolution/refresh rate mode to an existing virtual monitor.
+    RemoveMode {
+        /// ID of the virtual monitor to add a mode to.
+        id: driver_ipc::Id,
+
+        /// The index of the mode to remove.
+        mode_index: usize,
+    },
+    /// Enable a virtual monitor.
+    Enable { id: driver_ipc::Id },
+    /// Disable one or more virtual monitors.
+    Disable { id: driver_ipc::Id },
+    /// Remove one or more virtual monitors.
+    Remove { id: Vec<driver_ipc::Id> },
+    /// Remove all virtual monitors.
+    RemoveAll,
+}
+
+fn main() -> eyre::Result<()> {
+    let args = Args::parse();
+
+    match args.command {
+        Command::List => {
+            let mut client = Client::connect()?;
+
+            let monitors = client.list()?;
+
+            if args.json {
+                let mut stdout = std::io::stdout().lock();
+                serde_json::to_writer_pretty(&mut stdout, &monitors)?;
+            } else {
+                if monitors.len() > 0 {
+                    println!("{}", "Virtual monitors".underline());
+                    for (i, monitor) in monitors.iter().enumerate() {
+                        if i > 0 {
+                            println!();
+                        }
+
+                        let name_label = lazy_format!(match (&monitor.name) {
+                            Some(name) => (" {}{name}{}", "[".dimmed(), "]".dimmed()),
+                            None => "",
+                        });
+                        let disabled_label = lazy_format!(if monitor.enabled => ""
+                        else =>
+                            (" {}", "(disabled)".red())
+                        );
+                        println!(
+                            "Monitor {}{name_label}{disabled_label}:",
+                            monitor.id.green(),
+                        );
+
+                        if monitor.modes.len() > 0 {
+                            for (index, mode) in monitor.modes.iter().enumerate() {
+                                let refresh_rate_labels = mode
+                                    .refresh_rates
+                                    .iter()
+                                    .map(|rate| lazy_format!("{}", rate.blue()))
+                                    .join_with("/");
+                                let refresh_rates = lazy_format!(if mode.refresh_rates.is_empty() =>
+                                    ("{}Hz", "?".red())
+                                else =>
+                                    ("{}Hz", refresh_rate_labels)
+                                );
+                                println!(
+                                    "{} Mode {index}: {}x{} @ {}",
```

Comment:
And we will no longer need to notify or care about mode indexes once `remove-modes` is addressed. 😁

Reply:
_Marking as resolved, this will be handled in a follow-up_